File: tp-arithmetic-java/src/test/java/fr/isima/controller/StringControllerTest.java
package fr.isima.controller;

import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;


public class StringControllerTest {

    private StringController stringController = new StringController();

    @Test
    public void concat_ABCandDEF_returnsABCDEF() {
        assertThat(this.stringController.join("ABC", "DEF")).isEqualTo("ABCDEF");
    }

    @Test
    public void substringBefore_42AvantWithPatternAv_returns42() {
        assertThat(this.stringController.substringBefore("42Avant", "Av")).isEqualTo("42");
    }

    @Test
    public void substringAfter_Apres42WithPatternres_returns42() {
        assertThat(this.stringController.substringAfter("Apres42", "res")).isEqualTo("42");
    }

}

File: tp-arithmetic-kotlin/src/test/kotlin/fr/isima/controller/StringControllerTest.kt
package fr.isima.controller

import org.junit.Test

import org.assertj.core.api.Assertions.assertThat


class StringControllerTest {

    private val stringController = StringController()

    @Test
    fun concat_ABCandDEF_returnsABCDEF() {
        assertThat(this.stringController.join("ABC", "DEF")).isEqualTo("ABCDEF")
    }

    @Test
    fun substringBefore_42AvantWithPatternAv_returns42() {
        assertThat(this.stringController.substringBefore("42Avant", "Av")).isEqualTo("42")
    }

    @Test
    fun substringAfter_Apres42WithPatternres_returns42() {
        assertThat(this.stringController.substringAfter("Apres42", "res")).isEqualTo("42")
    }

}